spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -113,15 +113,42 @@
 
   describe 'GET /api/saml/auth' do
     context 'with LOA3 but the identity is already verified' do
+      before do
+        allow_any_instance_of(ServiceProvider).to receive(:attribute_bundle).and_return(
+          %w(first_name last_name ssn zipcode)
+        )
+      end
+
       it 'does not redirect the user to the IdV URL' do
         user = create(:user, :signed_up)
+        user.profiles.create(verified_at: Time.current, active: true, activated_at: Time.current)
         generate_saml_response(user, loa3_saml_settings)
 
-        user.profiles.create(verified_at: Time.current, active: true, activated_at: Time.current)
+        expect(response).to_not be_redirect
+      end
 
+      it 'contains verified attributes' do
+        user = create(:user, :signed_up)
+        user.profiles.create(
+          verified_at: Time.current,
+          active: true,
+          activated_at: Time.current,
+          first_name: 'Some',
+          last_name: 'One',
+          ssn: '666666666',
+          zipcode: '12345'
+        )
         generate_saml_response(user, loa3_saml_settings)
 
-        expect(response).to_not be_redirect
+        xpath = '//ds:Attribute[@Name="address1"]'
+        expect(decrypted_saml_response.at(xpath, ds: Saml::XML::Namespaces::ASSERTION)).to be_nil
+
+        %w(first_name last_name ssn zipcode).each do |attr|
+          xpath = %(//ds:Attribute[@Name="#{attr}"])
+          node = decrypted_saml_response.at(xpath, ds: Saml::XML::Namespaces::ASSERTION)
+          node_value = node.children.children.to_s
+          expect(node_value).to eq(user.active_profile[attr.to_sym])
+        end
       end
     end
```

Comment:
For completeness, what do you think about adding a test that verifies that the controller calls `AttributeAsserter#build`?